spec/ddtrace/contrib/faraday/middleware_spec.rb

```diff
@@ -242,4 +275,48 @@
       expect(request_span.service).to eq(service_name)
     end
   end
+
+  context 'configuration override' do
+    subject(:response) { client.get('/success') }
+
+    context 'with global configuration' do
+      let(:global_setting) { spy(proc {}) }
+      let(:configuration_options) { super().merge(error_handler: global_setting) }
```

Comment:
Using a real configuration option (`error_handler` here, instead of a test one like `fake_config`), allows for a much cleaner test, as we have safeguards in place to drop unknown configurations, make it hard to test with arbitrary configuration keys without a large amount of mocking.